engine/src/main/scala/cromwell/services/EngineMetadataServiceActor.scala

```diff
@@ -49,7 +54,9 @@ case class EngineMetadataServiceActor(serviceConfig: Config, globalConfig: Confi
     val sndr = sender()
     dataAccess.getWorkflowStatus(id) onComplete {
       case Success(Some(s)) => sndr ! StatusLookupResponse(id, s)
-      case Success(None) => sndr ! StatusLookupNotFound(id)
+      case Success(None) =>
+        val message = if (workflowExistenceCache.contains(id)) StatusLookupResponse(id, WorkflowSubmitted) else StatusLookupNotFound(id)
```

Comment:
This is reading mutable state in a callback which is usually dodgy for an actor, but it is a var/immutable which maybe is ok?  I can add a layer of messaging indirection to remove that layer of dodgy.